### DISCUSSION TITLE
Teritori integration

### DISCUSSION THREAD

**Discussion by D4ryl00** [berty/weshnet#54] [General]
Here is a conversation about the integration of weshnet in the Teritori project.

**Comment by costinberty** [comment]
hi

**Comment by sakul-budhathoki** [comment]
Hi @D4ryl00, @costinberty 
 we are integrating weshnet in Teritori. 
Teritori app is built using react-native expo. 
Will it work on the react-native expo app? Or, do we have to use go packages to make it work on mobile app.

**Reply by D4ryl00** [reply]
Hi @sakul-budhathoki, I don't know expo very well, but can you integrate an external framework into expo?

**Reply by sakul-budhathoki** [reply]
No, we can't integrate the external framework with the expo; we have decided to eject from expo and then integrate weshnet. Thanks.

**Comment by D4ryl00** [comment]
Ok, just in case have you seen that yet? https://docs.expo.dev/workflow/customizing/
There are two methods to add custom native code, development builds and prebuild (aka eject). I don't know the first one, but I know in Berty the team decided to eject expo too when they started the project in 2018.

**Reply by iuricmp** [reply]
I would try to avoid Expo ejection until the last possibility.
You could explore a way to use/create the Weshnet as an [Expo Module](https://docs.expo.dev/modules/overview)

**Comment by n0izn0iz** [comment]
hello, we are actually starting the weshnet integration in teritori

first target is teritori web + weshnet local cli daemon so we can start working asap, most of the work will be done by @sakul-budhathoki 

I needed to add a few things in weshnet repo to bootstrap him, I opened a PR with the changes for easier tracking here https://github.com/berty/weshnet/pull/70

the teritori PR is https://github.com/TERITORI/teritori-dapp/pull/609

**Reply by gfanton** [reply]
You should put the content of this PR directly in your repository. weshnet is designed to be used as a library. As for the generation part, you can utilize the Buf tool to download protobuf files directly.
